.github/workflows/check-sorbet-typing-mode.yml

```
@@ -0,0 +1,68 @@
+name: Sorbet
+on: # yamllint disable-line rule:truthy
+  pull_request:
+    branches: [main]
+    paths:
+      - '*/lib/dependabot/**/*.rb'  # Trigger the workflow only for Ruby files
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [main]  # Adjust this to the branch you want for push events
+    paths:
+      - '*/lib/dependabot/**/*.rb'  # Trigger the workflow only for Ruby files
+
+permissions:
+  contents: read
+
+jobs:
+  check-sorbet-typing-mode:
+    name: Sorbet Typing Mode Check
+    runs-on: ubuntu-latest
```

nit: here it doesn't really matter, but best practice is to set permissions on a per-job level...

```suggestion
jobs:
  check-sorbet-typing-mode:
    name: Sorbet Typing Mode Check
    runs-on: ubuntu-latest

    permissions:
      contents: read
```